src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +149,1026 @@ static DamageTable D_80A63CE0 = {
+    if (Actor_ProcessTalkRequest(&this->picto.actor, &play->state)) {
+        player->stateFlags1 &= ~PLAYER_STATE1_20;
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_TALK_TWIST);
+        this->forceEyesShut = false;
+        this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+        Message_StartTextbox(play, 0x1D17, &this->picto.actor);
+        this->prevTextId = 0x1D17;
+        this->picto.actor.flags &= ~ACTOR_FLAG_10000;
+        EnBal_SetupTalk(this);
+    } else {
+        func_800B85E0(&this->picto.actor, play, 200.0f, PLAYER_IA_MINUS1);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62B78.s")
+void EnBal_UpdateHead(EnBal* this, PlayState* play) {
+    Player* player;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62CF0.s")
+    if (this->watchTarget == EN_BAL_WATCHTRG_PLAYER) {
+        if ((this->skelAnime.animation != &gTingleWaitAnim) && (this->skelAnime.animation != &gTingleSpinAnim)) {
+            Actor_TrackPlayer(play, &this->picto.actor, &this->headRot, &this->torsoRot, this->picto.actor.focus.pos);
+        } else {
+            Actor_TrackNone(&this->headRot, &this->torsoRot);
+        }
+    } else if (this->watchTarget == EN_BAL_WATCHTRG_FAIRY) {
+        Vec3f tatlPos;
+        player = GET_PLAYER(play);
+        if (player->tatlActor != NULL) {
+            tatlPos = player->tatlActor->world.pos;
+            Actor_TrackPoint(&this->picto.actor, &tatlPos, &this->headRot, &this->torsoRot);
+        }
+    } else {
+        Actor_TrackNone(&this->headRot, &this->torsoRot);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62DCC.s")
+    if (this->forceEyesShut == true) {
+        this->eyeTexIndex = EN_BAL_EYETEX_CLOSED;
+    } else {
+        if (this->eyeTimer > 3) {
+            this->eyeTimer--;
+        } else {
+            if (this->eyeTimer != 0) {
+                this->eyeTexIndex = EN_BAL_EYETEX_CLOSED;
+                this->eyeTimer--;
+            } else {
+                this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+                this->eyeTimer = 60;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62ED0.s")
+void EnBal_UpdateCollision(EnBal* this, PlayState* play) {
+    if ((this->actionFunc == EnBal_FloatIdle) || ((this->actionFunc == EnBal_FloatUpToSky) && (this->timer > 50))) {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+    }
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62FAC.s")
+void EnBal_CheckBalloonPopped(EnBal* this, PlayState* play) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_CLEAR_TAG, this->picto.actor.world.pos.x,
+                    this->picto.actor.world.pos.y + 100.0f, this->picto.actor.world.pos.z, 0xFF, 0xFF, 0xC8, 2);
```

Comment:
```suggestion
        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_CLEAR_TAG, this->picto.actor.world.pos.x,
                    this->picto.actor.world.pos.y + 100.0f, this->picto.actor.world.pos.z, 0xFF, 0xFF, 0xC8, CLEAR_TAG_POP);
```
Note that you might need to adjust this once https://github.com/zeldaret/mm/pull/1308 is merged

Reply:
Bump with #1308 merged now, you will want `CLEAR_TAG_PARAMS(CLEAR_TAG_POP)`